server/src/main/scala/za/co/absa/atum/server/api/service/PartitioningServiceImpl.scala

```diff
@@ -38,7 +39,36 @@ class PartitioningServiceImpl(partitioningRepository: PartitioningRepository)
   ): IO[ServiceError, Either[StatusException, Unit]] = {
     repositoryCallWithStatus(
       partitioningRepository.createOrUpdateAdditionalData(additionalData), "createOrUpdateAdditionalData"
-    )
+    ).mapError(error => ServiceError(error.message))
+  }
+
+  override def getPartitioningMeasures(partitioning: PartitioningSubmitDTO): IO[ServiceError, Seq[MeasureDTO]] = {
+    partitioningRepository.getPartitioningMeasures(partitioning)
+      .mapError { case DatabaseError(message) =>
+        ServiceError(s"Failed to retrieve partitioning measures': $message")
+      }
+  }
+
+  override def getPartitioningAdditionalData(partitioning: PartitioningSubmitDTO): IO[ServiceError, AdditionalDataDTO] = {
+    partitioningRepository.getPartitioningAdditionalData(partitioning)
+      .mapError { case DatabaseError(message) =>
+        ServiceError(s"Failed to retrieve partitioning additional data': $message")
+      }
+  }
+
+  override def returnAtumContext(partitioningSubmitDTO: PartitioningSubmitDTO): IO[ServiceError, AtumContextDTO] = {
+    for {
+      partitioning <- createPartitioningIfNotExists(partitioningSubmitDTO)
+        .flatMap {
+          case Left(_) => ZIO.fail(ServiceError("Failed to create or retrieve partitioning"))
+          case Right(_) => ZIO.succeed(partitioningSubmitDTO)
```

Comment:
Okay. But btw, `createPartitioningIfNotExists` return Unit, and it could return Long (partitioning ID) - but I guess that we don't need it.